bson/src/main/org/bson/codecs/UuidRepresentationOverridingCodec.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs;
+
+import org.bson.UuidRepresentation;
+
+/**
+ * A marker interface for {@code Codec} implementations that can derive a new instance of itself that overrides the default
+ * {@code UuidRepresentation}.
+ *
+ * @param <T> the value type
+ * @since 3.12
```

Comment:
The specification says that "support for this specification can be added to a minor release, but with several exceptions", one of which is:

> Subtype 4 BSON Binary values should continue to be decoded to native UUID values regardless of UUID representation

So for 3.12 we can follow this advice and change the behavior in 4.0.  Does that seem sufficient?

Reply:
So for 3.12 we'd keep the existing behaviour for both subtypes?

Reply:
Yes.  The most recent patch implements that behavior

Reply:
Let me amend that statement.  We'll keep the same behavior for subtype 4 in that JAVA_LEGACY uuid representation will result in UUID.  For subtype 3, we can also decode the other legacy uuid representations to UUID.  This isn't a backwards breaking change though, because the ability to set uuid representation at all is an opt-in feature.

Reply:
Cool, thanks for the clarification.